openpype/pipeline/anatomy.py

```diff
@@ -339,6 +335,59 @@ def replace_root_with_env_key(self, filepath, template=None):
         return rootless_path.format(**data)
 
 
+class Anatomy(BaseAnatomy):
+    _project_cache = {}
+
+    def __init__(self, project_name=None, site_name=None):
+        if not project_name:
+            project_name = os.environ.get("AVALON_PROJECT")
+
+        if not project_name:
+            raise ProjectNotSet((
+                "Implementation bug: Project name is not set. Anatomy requires"
+                " to load data for specific project."
+            ))
+
+        self._site_name = site_name
+        project_info = self.get_project_data_and_cache(project_name, site_name)
+
+        super(Anatomy, self).__init__(
+            project_info["project_doc"],
+            project_info["local_settings"]
+        )
+
+    @classmethod
+    def get_project_data_and_cache(cls, project_name, site_name):
```

Comment:
I would not cache `local_settings`, they're already cached in settings logic. This should just return project document.